### PR TITLE
Always use tweaked keys with taproot

### PR DESCRIPTION
**Pull request by xoloki**
When doing taproot signatures with distributed keys, it is necessary to always use tweaked keys even when there is no script spend path. This is because a malicious DKG actor can insert a hidden script spend if the key is not tweaked.

So for all taproot sign functions, the merkle root should be optional, but tweaking the keys is not. If there is no merkle root, then the tweak is the hash of the public key only.

### CODE REVIEW

File: src/v1.rs [hunk 1]
```diff
@@ -189,37 +189,28 @@ impl Party {
         }
     }
 
-    /// Sign `msg` with this party's share of the group private key, using the set of `sigers` and corresponding `nonces` with a precomputed `aggregate_nonce`
+    /// Sign `msg` with this party's share of the group private key, using the set of `sigers` and corresponding `nonces` with a precomputed `aggregate_nonce` and an optional tweak to the public key
```
Comment (jferrant) [inline src/v1.rs]: Don't think this comment is accurate anymore? Doesn't seem like the tweak is optional?

File: src/v1.rs [hunk 2]
```diff
@@ -189,37 +189,28 @@ impl Party {
...
-            id: self.id,
-            z_i: z,
-            key_ids: vec![self.id],
-        }
+        self.sign_precomputed_with_tweak(msg, signers, nonces, aggregate_nonce, &Scalar::from(0))
     }
 
-    /// Sign `msg` with this party's share of the group private key, using the set of `sigers` and corresponding `nonces` with a precomputed `aggregate_nonce` and a tweaked public key
-    pub fn sign_precomputed_tweaked(
+    /// Sign `msg` with this party's share of the group private key, using the set of `sigers` and corresponding `nonces` with a precomputed `aggregate_nonce` and an optional tweak to the public key
```
Comment (jferrant) [inline src/v1.rs]: I don't think the tweak is optional here either. Is it not a guaranteed tweak?

File: src/taproot.rs [hunk 3]
```diff
@@ -413,15 +287,15 @@ mod test {
...
-        let tweaked_public_key = compute::tweaked_public_key(&aggregate_public_key, &merkle_root);
-
-        let (nonces, sig_shares) =
-            test_helpers::sign_tweaked(&msg, &mut S, &mut rng, &tweaked_public_key);
-        let sig = match sig_agg.sign_merkle_root(&msg, &nonces, &sig_shares, &key_ids, &merkle_root)
-        {
-            Err(e) => panic!("Aggregator sign failed: {:?}", e),
-            Ok(sig) => sig,
-        };
+        let tweaked_public_key =
```
Comment (jferrant) [inline src/taproot.rs]: Might not be a bad idea to add a test with "none" for the Merkle Root.

File: src/v1.rs (hunk 1 above)
Comment (xoloki) [reply src/v1.rs]: Fixed

File: src/v1.rs (hunk 2 above)
Comment (xoloki) [reply src/v1.rs]: Comment fixed

File: src/taproot.rs [hunk 4]
```diff
@@ -213,112 +193,6 @@ mod test {
     use crate::{compute, traits::Signer, v1, v2};
     use rand_core::OsRng;
 
-    #[test]
```
Comment (jferrant) [inline src/taproot.rs:216]: Are these tests no longer valid? Just want to make sure they were deleted intentionally.

File: src/taproot.rs (hunk 3 above)
Comment (xoloki) [reply src/taproot.rs]: Done, for v1 and v2

**Review by jferrant — APPROVED** [review]
LGTM

File: src/taproot.rs (hunk 4 above)
Comment (xoloki) [reply src/taproot.rs:216]: Yeah, this file is now taproot only, so there's no need to do a non-taproot test.

We originally thought it would be okay to do schnorr without a tweaked key, but after investigating I realized that there was no way to safely do a distributed key without tweaking it.

**Comment by xoloki** [comment]
codecov is failing with a config issue:

[2023-09-05T17:09:04.026Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}

All other tests are passing so I'll go ahead and merge to unblock external users.
